paymentsheet/src/test/java/com/stripe/android/paymentsheet/PaymentSheetViewModelTest.kt

```diff
@@ -1071,6 +1072,30 @@ internal class PaymentSheetViewModelTest {
         }
     }
 
+    @Test
+    fun `On fail due to invalid deferred intent usage, should report with expected integration error`() = runTest {
+        val interceptor = FakeIntentConfirmationInterceptor().apply {
+            enqueueFailureStep(
+                cause = InvalidDeferredIntentUsageException(),
+                message = "An error occurred!",
+            )
+        }
+
+        val viewModel = createViewModel(
+            intentConfirmationInterceptor = interceptor,
+        )
+
+        viewModel.updateSelection(CARD_PAYMENT_SELECTION)
+        viewModel.checkout()
+
+        verify(eventReporter).onPaymentFailure(
+            paymentSelection = CARD_PAYMENT_SELECTION,
+            error = PaymentSheetConfirmationError.MerchantIntegration(
+                analyticsValue = "invalidDeferredIntentUsage",
+            ),
+        )
```

Comment:
We don't have a `FakeEventReporter` yet. I could add it in this PR or follow up with it.

Reply:
I added a no op one in a WIP PR. But a `RecordingEventReporter` would probably also be useful.

Reply:
I suppose we could also make the no op variant an open class, and just override the one method in tests like these to do the "recording"